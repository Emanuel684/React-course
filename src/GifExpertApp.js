import React, { useState } from 'react';

import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaulCategories = [] }) => {

    // const [categories, setCategories] = useState(['One Piace']);
    const [categories, setCategories] = useState(['One Piace']);


    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map(category => (
                        // <li key={category} >{category}</li>
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>

        </>
    )

}

export default GifExpertApp;