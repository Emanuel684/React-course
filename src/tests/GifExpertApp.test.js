import React from 'react';
import GifExpertApp from "../GifExpertApp";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
// import { useFetchGifs } from '../../hooks/useFetchGifs';
// jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en <GifExpertApp />', () => {

    const category = 'One Punch';


    test('Debe de mostrar <GifExpertApp /> correctamente', () => {

        // useFetchGifs.mockReturnValue({
        //     data: [],
        //     loading: true
        // });

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });
    

});