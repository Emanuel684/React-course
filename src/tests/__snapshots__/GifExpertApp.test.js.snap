// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piace"
      key="One Piace"
    />
  </ol>
</Fragment>
`;
